### PR TITLE
Add `FixFilePermissionsTask` to fix up `NULL` file permissions from SS3

### PR DESCRIPTION
**Pull request by bergice**
This just resets the view enum to `Inherit` for records that have it set to `NULL`.

This task will be combined with other file migration tasks at some point so we don't have too many of them floating around.

This should also just be necessary after upgrading from SS3 as it is a datafix.

### CODE REVIEW

**Comment by chillu** [comment]
This should be a subtask of `MigrateFileTask`, for the same reasons as thumbnail and shortcode migration subtasks. This is part of the overall "4.x file migration", but in some cases you just want to run those tasks in isolation. I don't want people doing 3.x -> 4.x upgrades to start off with running a whole bunch of arbitrary CLI commands to get started :) Can you please rewrite this to be a subtask, based on the work in https://github.com/silverstripe/silverstripe-assets/issues/235

File: src/Dev/FixFilePermissionsTask.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace SilverStripe\Dev\Tasks;
+
+use SilverStripe\Assets\File;
+use SilverStripe\Dev\BuildTask;
+use SilverStripe\ORM\Queries\SQLUpdate;
+use SilverStripe\Security\InheritedPermissionFlusher;
+
+/**
+ * Files imported from SS3 might end up with broken permissions if there is a case conflict.
```
Comment (chillu) [inline src/Dev/FixFilePermissionsTask.php]: Should mention that this fixes issues with https://github.com/silverstripe/silverstripe-secureassets

File: src/Dev/FixFilePermissionsTask.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Reset the permission of files that have a `NULL` CanViewType value.
+    You need to flush your cache after running this task via CLI.
+    ';
+
+    public function run($request)
+    {
+        SQLUpdate::create(
+            File::singleton()->baseTable(),
+            ['CanViewType' => 'Inherit'],
+            ['ISNULL(CanViewType)']
```
Comment (chillu) [inline src/Dev/FixFilePermissionsTask.php]: Should filter by `Folder` class name

File: src/Dev/FixFilePermissionsTask.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private static $segment = 'FixFilePermissionsTask';
+
+    protected $title = 'Fix File Permissions';
+
+    protected $description = '
+    Reset the permission of files that have a `NULL` CanViewType value.
+    You need to flush your cache after running this task via CLI.
+    ';
+
+    public function run($request)
```
Comment (chillu) [inline src/Dev/FixFilePermissionsTask.php]: Should have some logging - e.g. `Repaired X folders with broken CanViewType settings`. Running two raw SQL queries (one count, one update) is fine for that

File: tests/php/FixFilePermissionsTaskTest.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  parent:
+    Name: ParentFolder
+    CanViewType: NULL
+  subfolder:
+    Name: SubFolder
+    Parent: =>SilverStripe\Assets\Folder.parent
+SilverStripe\Assets\Image:
+  image1:
+    Name: myimage.jpg
+    CanViewType: LoggedInUsers
```
Comment (chillu) [inline tests/php/FixFilePermissionsTaskTest.yml]: Only folders can have `CanViewType`, at least when they're using https://github.com/silverstripe/silverstripe-secureassets. Files can't. While there could be other ways in which files have data in this column, it's not in scope here - we're only fixing something which a supported module broke, and then had downstream effects on core file migrations.

**Comment by bergice** [comment]
> This should be a subtask of `MigrateFileTask`, for the same reasons as thumbnail and shortcode migration subtasks. This is part of the overall "4.x file migration", but in some cases you just want to run those tasks in isolation. I don't want people doing 3.x -> 4.x upgrades to start off with running a whole bunch of arbitrary CLI commands to get started :) Can you please rewrite this to be a subtask, based on the work in #235

@chillu should we merge https://github.com/silverstripe/silverstripe-framework/pull/8924/files first or do you want me to cherry-pick/rebase on your branch?

**Comment by chillu** [comment]
I'd say rebase on that PR, rather than cherry pick. Don't let the https://github.com/silverstripe/silverstripe-framework/pull/8924 merge block you for now.

**Pull request merged**